BinaryEncoding.md

```diff
@@ -369,17 +347,19 @@ a `data_segment` is:
 
 ### Name section
 
-ID: `name`
+User-defined section string: `"name"`
 
-The names section does not change execution semantics and a validation error in
-this section does not cause validation for the whole module to fail and is
-instead treated as if the section was absent. The expectation is that, when a
-binary WebAssembly module is viewed in a browser or other development
+The names section does not change execution semantics, and thus is not allocated a section code.
+It is encoded as an unknown section (id `0`) with the first few payload bytes identifying this section as a string.
+Like all unknown sections, a validation error in this section does not cause validation of the module to fail.
+The expectation is that, when a binary WebAssembly module is viewed in a browser or other development
 environment, the names in this section will be used as the names of functions
 and locals in the [text format](TextFormat.md).
 
 | Field | Type | Description |
 | ----- | ---- | ----------- |
+| name_length | `varint7` | length of the "name" string = 4 |
```

Comment:
Do we expect all "unknown" sections begin with name_length and name_string or can an unknown section conceivably have a payload begin with something else? Maybe unknown is its own section category and needs to be defined and Name section is a subcategory. Admittedly, this may be overkill.